File: src/main/java/net/shtyftu/ubiquode/service/QuestException.java
package net.shtyftu.ubiquode.service;

/**
 * @author shtyftu
 */
abstract class QuestException extends Exception {

    QuestException() {
        super();
    }

    QuestException(String s) {
        super(s);
    }

    static class WrongStateException extends QuestException {

    }

    static class OnlyDeadlineLowScoreUserException extends QuestException {

    }

    static class UserUnableToLockException extends QuestException {

        UserUnableToLockException(String s) {
            super(s);
        }
    }
}

File: src/main/java/net/shtyftu/ubiquode/service/QuestException.kt
package net.shtyftu.ubiquode.service

/**
 * @author shtyftu
 */
internal abstract class QuestException : Exception {

    constructor() : super()

    constructor(s: String) : super(s)

    internal class WrongStateException : QuestException()

    internal class OnlyDeadlineLowScoreUserException : QuestException()

    internal class UserUnableToLockException(s: String) : QuestException(s)
}
